### PR TITLE
chore(deps): update Renovate to ignore patches

### LINKED ISSUE #5873: RenovateBot: disable patch updates
Currently RenovateBot has been pushing a lot of patch updates. This creates a huge burden on maintaining the dependencies when tests take more than 20 minutes to run and there's significantly high number of traffic.

The initial problem is that RenovateBot by default does not distinguish between patch and minor updates. Thus, since currently we have `major` and `minor` updates enabled: it enables RenovateBot to pick up on patches as well.

As desired we should fix this behavior and only update on major/minor releases.

### PR DESCRIPTION
**Pull request by dandhlee**
## Description

The problem with RenovateBot currently is that it actually treats `minor` and `patch` releases to be the same. See [separateMinorPatch](https://docs.renovatebot.com/configuration-options/#separateminorpatch) docs for more detail.

To mitigate this, we first include a rule to tell RenovateBot to separate and recognize between `minor` and `patch` releases. Then, we create a second rule to disable `patch` updates. This should now successfully let RenovateBot to only update on `major` and `minor` patches.

Fixes #5873 

Note: It's a good idea to open an issue first for discussion.

## Checklist
- [X] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.6` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] Please **merge** this PR for me once it is approved.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

**Comment by dandhlee** [comment]
Adding back the original snippet for specifying `major` and `minor` updates back for good faith :S

I'm not sure if specifying them will be required, but this should now be explicit to say

```
separate minor and patch releases
ignore patch releases
update on major and minor releases
```

rather than assuming that the third line (update on major and minor) is achieved by the second line.

**Review by busunkim96 — APPROVED** [review]

File: renovate.json [hunk 1]
```diff
@@ -9,6 +9,15 @@
...
+      "separateMinorPatch": true
+    },
+    {
+      "packagePatterns": ["pytest"],
+      "updateTypes": ["patch"],
+      "enabled": false
+    },
     {
       "packagePatterns": ["pytest"],
       "updateTypes": ["minor", "major"]
```
Comment (leahecole) [inline renovate.json:23]: DO we need to also remove this section? Or do these rules work with the existing pattern?

Comment (leahecole) [reply renovate.json:23]: nvm, i see the comment

**Review by leahecole — APPROVED** [review]
YAY

**Comment by dandhlee** [comment]
Hoping this works! I'll follow up with `boto3` since that one gets a daily update.

**Pull request merged**